docs/features/docker_auth.md

```diff
@@ -0,0 +1,52 @@
+# Authentication with Docker
+
+Sometimes the Docker images you use live in a private Docker registry. For that reason, _Testcontainers for Go_ gives you the ability to read the Docker configuration
+and retrieve the authentication for a given registry. To achieve it, _Testcontainers for Go_ will internally check, in this particular order:
+	
+1. the `DOCKER_AUTH_CONFIG` environment variable, unmarshalling the string value from its JSON representation and using it as the Docker config.
+2. the `DOCKER_CONFIG` environment variable, as an alternative path to the Docker config file.
+3. else it will load the default Docker config file, which lives in the user's home, e.g. `~/.docker/config.json`
+4. it will use the right Docker credential helper to retrieve the authentication (user, password and base64 representation) for the given registry.
+
+To understand how the Docker credential helpers work, please refer to the [official documentation](https://docs.docker.com/engine/reference/commandline/login/#credential-helpers).
+
+!!! info
+	_Testcontainers for Go_ uses [https://github.com/cpuguy83/dockercfg](https://github.com/cpuguy83/dockercfg) to retrieve the authentication from the credential helpers.
+
+To retrieve the credentials for a given registry, _Testcontainers for Go_ exposes the `AuthFromDockerConfig` function, which takes the registry name as a parameter and returns the authentication for that registry:
+
+```go
+auth, err := testcontainers.AuthFromDockerConfig("myregistry.com")
+if err != nil {
+	// do something with err
+}
+```
```

Comment:
You cannot resolve the credentials internally (automatic), based on the image name?

Reply:
I was wondering the same. In testcontainers-java, this process is transparent for the user (in the same way it is present for the user when using Docker CLI).

Reply:
I can include it automatically :)
> We decided to provide that capability instead of automatically discovering it, but I'd not be opposed to do it if desired.

I can imagine this workflow:
- if the user provided `req.RegistryCred`, use it
- else, extract the registry name from the image, and load them

Is this correct?

Reply:
For .NET we always extract the registry name from the image.

Reply:
.NET had a similar feature. With the help of various credential providers, I removed it. However, there are still occasional discussions about whether we should support it or not, as seen in this GitHub issue: https://github.com/testcontainers/testcontainers-dotnet/issues/755. Since I understand how the helper and other components work, I prefer the approach of automatically detecting the configuration. However, I can understand that it may be difficult for users to grasp. Although in most environments, users probably do not notice it because it just works.